packages/wrangler/src/ai/fetcher.ts

```diff
@@ -16,12 +16,14 @@ export default function (env) {
 export async function AIFetcher(request: Request): Promise<Response> {
 	const accountId = await getAccountId();
 
-	request.headers.delete("Host");
-	request.headers.delete("Content-Length");
+	const reqHeaders = new Headers(request.headers);
+	reqHeaders.delete("Host");
+	reqHeaders.delete("Content-Length");
+	reqHeaders.set("X-Forwarded-Host", request.url);
```

Comment:
Just to be clear I am not sure that `X-Forwarded-Host` is the correct thing here. It is what I was thinking of but I think is only for hosts not full URLs.

Do you have a requirement for this header on the server?
What is the header it is looking for?

Reply:
i'm writing the server part as well, so we are free to decide what we like best 😄 
the only requirement i have is for this local fetcher to send the full path and query string to the server, so any header name does the job